src/cockpit/channel.py

```diff
@@ -397,3 +397,35 @@ def do_data(self, data):
             data = bytes(data)
 
         self.receive_queue.put_nowait(data)
+
+
+class GeneratorChannel(Channel):
+    """A trivial Channel subclass for sending data from a generator with flow control.
+
+    Calls the .get_data() generator with the options from the open message and
+    sends the data which it yields.  If the generator returns a value it will
+    be used for the close message.
+    """
+    __generator: Generator[bytes, int, Optional[Dict[str, object]]]
+
+    def get_data(self, size_hint: int, options: dict[str, object]) -> Generator[bytes, int, Optional[Dict[str, object]]]:
```

Comment:
Could the return type get its own type name? I think mypy/typing allow that. It's not quite trivial, and gets repeated in fsread1.

Reply:
It could, although I don't consider this particular type to be so complicated.  Generators are just like that :sweat_smile:

Reply:
Heh, ok -- not important either way, feel free to just resolve.

Reply:
I added it.